File: app/src/main/java/com/tutv/android/domain/Season.java
package com.tutv.android.domain;

import androidx.room.ColumnInfo;
import androidx.room.Entity;
import androidx.room.ForeignKey;
import androidx.room.Ignore;
import androidx.room.PrimaryKey;

import java.util.List;

@Entity(tableName = "season", foreignKeys = {@ForeignKey(entity = Series.class, parentColumns = "series_id", childColumns = "series_id")})
public class Season {

    @PrimaryKey
    @ColumnInfo(name = "season_id")
    private int id;

    private int number;

    @Ignore List<Episode> episodes;
    @Ignore boolean isExpanded = false;

    @ColumnInfo(name = "series_id") private int seriesId;

    public int getId() {
        return id;
    }

    public void setId(int id) {
        this.id = id;
    }

    public int getNumber() {
        return number;
    }

    public void setNumber(int number) {
        this.number = number;
    }

    public List<Episode> getEpisodes() {
        return episodes;
    }

    public void setEpisodes(List<Episode> episodes) {
        this.episodes = episodes;
    }

    public boolean isExpanded() {
        return isExpanded;
    }

    public void setExpanded(boolean expanded) {
        isExpanded = expanded;
    }

    public int getSeriesId() {
        return seriesId;
    }

    public void setSeriesId(int seriesId) {
        this.seriesId = seriesId;
    }
}

File: app/src/main/java/com/tutv/android/domain/Season.kt
package com.tutv.android.domain

import androidx.room.*

@Entity(
    tableName = "season",
    foreignKeys = [ForeignKey(
        entity = Series::class,
        parentColumns = arrayOf("series_id"),
        childColumns = arrayOf("series_id")
    )]
)
data class Season(@PrimaryKey @ColumnInfo(name = "season_id") var id: Int = 0,
                              @ColumnInfo(name = "series_id") var seriesId: Int = 0,
                                                      @Ignore var episodes: List<Episode>? = null,
                                                      @Ignore var isExpanded: Boolean = false,
                                                              var number: Int = 0)